packages/turf-transform-rotate/index.js

```diff
@@ -0,0 +1,114 @@
+ * @param {number} [xRotation=0] angle of rotation, respect to the horizontal (x,y) plain, about the x axis;
+ * @param {number} [yRotation=0] angle of rotation, respect to the horizontal (x,y) plain, about the y axis;
+ * @returns {Feature<any>} the rotated GeoJSON feature
+ * @example
+ * var poly = turf.polygon([[0,29],[3.5,29],[2.5,32],[0,29]]);
+ * var rotatedPoly = turf.rotate(poly, 100, 35);
+ *
+ * //addToMap
+ * var addToMap = [rotatedPoly];
+ */
+module.exports = function (geojson, angle, pivot, xRotation, yRotation) {
+    // Input validation
+    if (!geojson) throw new Error('geojson is required');
+    if (angle === undefined || angle === null || isNaN(angle)) throw new Error('angle is required');
+    if (pivot && !Array.isArray(pivot) && pivot.geometry.type !== 'Point' && pivot.type !== 'Point') throw new Error('invalid pivot');
+    // if (yRotation && typeof yRotation !== 'number' && isNaN(yRotation)) throw new Error('yRotation is not a number');
+    // if (xRotation && typeof xRotation !== 'number' && isNaN(xRotation)) throw new Error('xRotation is not a number');
+
+    if (geojson.type !== 'Feature') geojson = feature(geojson);
+
+    // shortcut no-rotation
+    if (angle === 0 && yRotation === 0) return geojson;
+
+    if (!pivot) {
+        pivot = centroid(geojson);
+    } else if (Array.isArray(pivot)) {
+        pivot = point(pivot);
+    }
+
+    // copy properties to avoid reference issues
+    var properties = {};
+    Object.keys(geojson.properties).forEach(function (key) {
+        properties[key] = geojson.properties[key];
+    });
+
+    var rotatedCoords;
+
+    var type = geojson.geometry.type;
+    switch (type) {
+    case 'Point':
+        if (!pivot) return geojson;
+        rotatedCoords = rotate([getCoords(geojson)], angle, pivot);
+        return point(rotatedCoords[0], properties);
+    case 'MultiPoint':
+        rotatedCoords = rotate(getCoords(geojson), angle, pivot);
+        return multiPoint(rotatedCoords, properties);
+    case 'LineString':
+        rotatedCoords = rotate(getCoords(geojson), angle, pivot);
+        return lineString(rotatedCoords, properties);
+    case 'MultiLineString':
+        rotatedCoords = getCoords(geojson).map(function (lineCoords) {
+            return rotate(lineCoords, angle, pivot);
+        });
+        return multiLineString(rotatedCoords, properties);
+    case 'Polygon':
+        rotatedCoords = getCoords(geojson).map(function (ringCoords) {
+            return rotate(ringCoords, angle, pivot);
+        });
+        return polygon(rotatedCoords, properties);
+    case 'MultiPolygon':
+        rotatedCoords = getCoords(geojson).map(function (polyCoords) {
+            return polyCoords.map(function (ringCoords) {
+                return rotate(ringCoords, angle, pivot);
+            });
+        });
+        return multiPolygon(rotatedCoords, properties);
+    default:
+        throw new Error('geometry ' + type + ' is not supported');
```

Comment:
Impossible to hit this `default` switch statement.

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Statements/switch
> If no default clause is found, the program continues execution at the statement following the end of switch.